UndertaleModLib/Decompiler/Decompiler.cs

```diff
@@ -219,18 +221,33 @@ public override string ToString()
                 if (AssetType == AssetIDType.Color && Value is IFormattable)
                 {
                     uint val = Convert.ToUInt32(Value);
-                    return "0x" + ((IFormattable)Value).ToString(val > 0xFFFFFF ? "X8" : "X6", CultureInfo.InvariantCulture);
+                    return (isGMS2 ? "0x" : "$") + ((IFormattable)Value).ToString(val > 0xFFFFFF ? "X8" : "X6", CultureInfo.InvariantCulture);
                 }
 
                 if (AssetType == AssetIDType.KeyboardKey)
                 {
                     int val = Convert.ToInt32(Value);
+                    if (val >= 0 && Enum.IsDefined(typeof(EventSubtypeKey), (uint)val))
+                    {
+                        bool isAlphaNumeric = val >= (int)EventSubtypeKey.Digit0 && val <= (int)EventSubtypeKey.Z;
+                        return isAlphaNumeric ? "ord(\"" + (char)val + "\")" : ((EventSubtypeKey)val).ToString(); // Either return the key enum, or the right alpha-numeric key-press.
+                    }
+
                     if (!Char.IsControl((char)val) && !Char.IsLower((char)val)) // The special keys overlay with the uppercase letters (ugh)
-                        return "'" + (char)val + "'";
-                    if (val > 0 && Enum.IsDefined(typeof(EventSubtypeKey), (uint)val))
-                        return ((EventSubtypeKey)val).ToString();
+                        return ((char)val) == '\'' ? (isGMS2 ? "\"\\\"\"" : "'\"'")
+                            : (((char) val) == '\\' ? (isGMS2 ? "\"\\\\\"" : "\"\\\"")
+                            : "\"" + (char)val + "\"");
                 }
 
+                if (Value is float) // Prevents scientific notation by using high bit number.
+                    return ((decimal) ((float) Value)).ToString(CultureInfo.InvariantCulture);
```

Comment:
Technically "F99" will give you 99 0s, buuuuuut it makes more of a mess than this, because you have to get rid of trailing zeros, then handle the decimal point, and it can get really messy. This seemed like the cleanest fix.